frontend/src/hooks/useGame.ts

```diff
@@ -377,5 +401,15 @@ export const useGame = create<State>((set, get) => ({
                 mySecurity: updatedSecurity
             };
         });
+    },
+
+    setMemory: (memory: number) => {
+        set(state => {
+            return {
+                ...state,
+                myMemory: memory,
+                opponentMemory: -memory
```

Comment:
if you implement my state change suggestions, dont forget to refactor those and others in your store